packages/web/src/components/gcds-stepper/test/gcds-stepper.spec.tsx

```diff
@@ -22,13 +29,92 @@ describe('gcds-stepper', () => {
   it('renders steps', async () => {
     const page = await newSpecPage({
       components: [GcdsStepper],
-      html: `<gcds-stepper current-step="2" total-steps="6"></gcds-stepper>`,
+      html: `<gcds-stepper current-step="2" total-steps="6">Section title</gcds-stepper>`,
     });
     expect(page.root).toEqualHtml(`
       <gcds-stepper current-step="2" total-steps="6">
         <mock:shadow-root>
-          <gcds-heading tag="h6" class="gcds-stepper" margin-top="0" margin-bottom="300">Step 2 of 6</h6>
+          <gcds-heading tag="h2" class="gcds-stepper" margin-top="0" margin-bottom="300">
+            <span class="gcds-stepper__steps">
+              Step 2 of 6
+              <gcds-sr-only> : </gcds-sr-only>
+            </span>
+            <slot></slot
+          </gcds-heading>
         </mock:shadow-root>
+        Section title
+      </gcds-stepper>
+    `);
+  });
+
+  /**
+   * Renders as a heading 1
+   */
+  it('renders steps', async () => {
+    const page = await newSpecPage({
+      components: [GcdsStepper],
+      html: `<gcds-stepper current-step="2" total-steps="6">Section title</gcds-stepper>`,
+    });
+    expect(page.root).toEqualHtml(`
+      <gcds-stepper current-step="2" total-steps="6">
+        <mock:shadow-root>
+          <gcds-heading tag="h2" class="gcds-stepper" margin-top="0" margin-bottom="300">
+            <span class="gcds-stepper__steps">
+              Step 2 of 6
+              <gcds-sr-only> : </gcds-sr-only>
+            </span>
+            <slot></slot
+          </gcds-heading>
+        </mock:shadow-root>
+        Section title
+      </gcds-stepper>
+    `);
+  });
+
+  /**
+   * Renders as a heading 2
+   */
+  it('renders steps', async () => {
+    const page = await newSpecPage({
+      components: [GcdsStepper],
+      html: `<gcds-stepper current-step="2" total-steps="6" tag="h2">Section title</gcds-stepper>`,
+    });
+    expect(page.root).toEqualHtml(`
+      <gcds-stepper current-step="2" total-steps="6" tag="h2">
+        <mock:shadow-root>
+          <gcds-heading tag="h2" class="gcds-stepper" margin-top="0" margin-bottom="300">
+            <span class="gcds-stepper__steps">
+              Step 2 of 6
+              <gcds-sr-only> : </gcds-sr-only>
+            </span>
+            <slot></slot
+          </gcds-heading>
+        </mock:shadow-root>
+        Section title
+      </gcds-stepper>
+    `);
+  });
+
+  /**
+   * Renders as a heading 3
+   */
+  it('renders steps', async () => {
```

Comment:
And also add another test to add a heading that's not a simple text (i.e. inside another tag)?